tw2023_wallet/Services/OID/OpenIdProvider.swift

```diff
@@ -851,6 +825,37 @@ func selectDisclosure(sdJwt: String, presentationDefinition: PresentationDefinit
     return nil
 }
 
+private func extractFieldKeys(
+    from sourcePayload: [String: String], using inputDescriptor: InputDescriptor
+) -> [(String, Bool)] {
+    guard let fields = inputDescriptor.constraints.fields else { return [] }
+    return fields.flatMap { field in
+        let optional = field.optional ?? false
+        return field.path.compactMap { jsonPath in
+            let key = String(jsonPath.dropFirst(2))  // "$."を削除
+            return sourcePayload.keys.contains(key) ? (key, optional) : nil
+        }
+    }
+}
+
+private func createDisclosureWithOptionality(
+    from disclosures: [Disclosure], with fieldKeys: [(String, Bool)]
+) -> [DisclosureWithOptionality] {
+    return disclosures.map { disclosure in
+        guard let dkey = disclosure.key else {
+            return DisclosureWithOptionality(
+                disclosure: disclosure, isSubmit: false, optional: false)
```

Comment:
> disclosure: disclosure, isSubmit: false, optional: false)

It seems a little odd that the optional value false is applied to claims that are not required to be disclosed.

Reply:
Although this depends on future UI requirements changes, under the current assumption, I am considering displaying a list of non-disclosed claims on the screen for the user’s peace of mind. Therefore, the `false false` combination is necessary. It is used in the following processes
https://github.com/moratori/OWND-Wallet-iOS/blob/8e728824440c2ba1860e9f61c16c934fcd444b14/tw2023_wallet/Feature/Credentials/ViewModels/CredentialDetailViewModel.swift#L49-L50

Upon reviewing the code following your comments, I noticed that the above intent was not clearly conveyed through the code, so I have updated the function names and variable names accordingly. I apologize for the inconvenience, but I would appreciate it if you could review the changes.
https://github.com/moratori/OWND-Wallet-iOS/commit/8e728824440c2ba1860e9f61c16c934fcd444b14

Reply:
Sorry if I misunderstood your comment. 

(Maybe it would be better to return `nil` in the `return` below and change the `map` above to `compactMap`?)
https://github.com/moratori/OWND-Wallet-iOS/blob/8e728824440c2ba1860e9f61c16c934fcd444b14/tw2023_wallet/Services/OID/OpenIdProvider.swift#L864-L865

Reply:
@ryosuke-wakaba Thank you for letting me speak to you directly earlier. I understand the point you made. As you said, currently there is an issue where `optional` being `false` seems to mean `required`.

The actual processing is using this value to determine whether "the user can change whether to send or not".
As such, I have changed the name to an appropriate one based on the processing content: `optional` -> `isUserSelectable`

https://github.com/moratori/OWND-Wallet-iOS/commit/13bc1561edb987cfdfb324e4782e39f1b847d5a5